src/connectedk8s/azext_connectedk8s/_validators.py

```diff
@@ -17,3 +27,31 @@ def example_name_or_id_validator(cmd, namespace):
                 type='storageAccounts',
                 name=namespace.storage_account
             )
+
+
+def check_and_validate_custom_registry_arguments(container_registry_repository=None, container_registry_username=None,
+                                                 container_registry_password=None, container_registry_agent_version=None,
+                                                 auto_upgrade=None):
+
+    if container_registry_repository is None:
+        if container_registry_username is not None:
+            logger.warning("Ignoring --registry-username as --registry-repository is not provided")
```

Comment:
I think we should throw error here because, if someone does not pass 'container_registry_repository' and passes both 'container_registry_username' and 'container_registry_password', the code will end up doing a helm login to mcr with the creds passed by the customer

Reply:
If the 'container_registry_repository' is not passed and credentials are passed then it will ignore those credentials. It will not use those credentials for MCR. Its an expected behavior also we are giving user a warning message so it should be good

Reply:
okay, then i probably missed where we are ignoring those params. From the flow, it seemed like we will end up doing helm login to mcr with pcr creds